apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestionTable.tsx

```diff
@@ -0,0 +1,781 @@
+    ];
+
+    // Only add filters for columns that exist
+    if (aiGradingMode && instanceQuestionGroups.length > 0) {
+      filters.push({ id: 'instance_question_group_name', value: submissionGroupFilter });
+    }
+    if (aiGradingMode) {
+      filters.push({ id: 'rubric_difference', value: aiAgreementFilter });
+    }
+    // Filter by rubric items if any are selected
+    if (rubricData && rubricData.rubric_items.length > 0 && rubricItemsFilter.length > 0) {
+      filters.push({ id: 'rubric_grading_item_ids', value: rubricItemsFilter });
+    }
+    if (assessmentQuestion.max_auto_points && assessmentQuestion.max_auto_points > 0) {
+      filters.push({ id: 'auto_points', value: autoPointsFilter });
+    }
+    if (!aiGradingMode) {
+      filters.push({ id: 'score_perc', value: scoreFilter });
+    }
+
+    return filters;
+  }, [
+    gradingStatusFilter,
+    assignedGraderFilter,
+    gradedByFilter,
+    submissionGroupFilter,
+    aiAgreementFilter,
+    rubricItemsFilter,
+    manualPointsFilter,
+    autoPointsFilter,
+    totalPointsFilter,
+    scoreFilter,
+    aiGradingMode,
+    instanceQuestionGroups.length,
+    assessmentQuestion.max_auto_points,
+    rubricData,
+  ]);
+
+  // Create columns using the extracted function
+  const columns = useMemo(
+    () =>
+      createColumns({
+        aiGradingMode,
+        instanceQuestionGroups,
+        groupWork,
+        assessmentQuestion,
+        authzDataHasCourseInstancePermissionEdit:
+          authzData.has_course_instance_permission_edit ?? false,
+        urlPrefix,
+        csrfToken,
+        assessmentId,
+        createCheckboxProps,
+      }),
+    [
+      aiGradingMode,
+      instanceQuestionGroups,
+      groupWork,
+      assessmentQuestion,
+      authzData,
+      urlPrefix,
+      csrfToken,
+      assessmentId,
+      createCheckboxProps,
+    ],
+  );
+
+  const allColumnIds = columns.map((col) => col.id).filter((id) => typeof id === 'string');
+  const defaultColumnVisibility = Object.fromEntries(
+    allColumnIds.map((id) => [
+      id,
+      // Hide by default: user_or_group_name, uid, points, rubric_grading_item_ids
+      !['user_or_group_name', 'uid', 'points', 'rubric_grading_item_ids'].includes(id),
+    ]),
+  );
+
+  const [columnVisibility, setColumnVisibility] = useQueryState(
+    'columns',
+    parseAsColumnVisibilityStateWithColumns(allColumnIds).withDefault(defaultColumnVisibility),
+  );
+
+  // Update column visibility when AI grading mode changes
+  useEffect(() => {
+    void setColumnVisibility((prev) => ({
+      ...prev,
+      // Hide these columns in AI grading mode
+      requires_manual_grading: !aiGradingMode,
+      assigned_grader_name: !aiGradingMode,
+      score_perc: !aiGradingMode,
+      // Show these columns in AI grading mode
+      instance_question_group_name: aiGradingMode,
+      rubric_difference: aiGradingMode,
+    }));
+  }, [aiGradingMode, setColumnVisibility]);
```

Comment:
This might be responsible for the weird flash of the wrong content when toggling modes? This shouldn't be necessary anyways; we'll show the correct columns after the page reloads.

EDIT: I removed this code and confirmed that that's the case.